src/PKSim.Core/Services/CompoundPKContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+using PKSim.Core.Model;
+
+namespace PKSim.Core.Services
+{
+   public class CompoundPKContext
+   {
+      private readonly ICache<string, CompoundPK> _compoundPKCache;
+
+      public CompoundPKContext()
+      {
+         _compoundPKCache = new Cache<string, CompoundPK>(x => x.CompoundName, x => new CompoundPK());
+      }
+
+      public CompoundPK CompoundPKFor(string compoundName)
+      {
+         if (!_compoundPKCache.Contains(compoundName))
+            _compoundPKCache.Add(new CompoundPK { CompoundName = compoundName });
+
+         return _compoundPKCache[compoundName];
+      }
+
+      public void AddCompoundPK(CompoundPK compoundPK)
+      {
+         _compoundPKCache[compoundPK.CompoundName] = compoundPK;
```

Comment:
hum... why are you passing the key here? Can it exist already? Otherwise do _compoundPKCache.add(